src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -23,38 +37,191 @@
     KShooterEncoder = shooter_motorController.getEncoder();
-    KShooterController.setP(5e-4);
-    KShooterController.setI(6e-7);
-    KShooterController.setD(0.0);
+    
+    DistanceArray[0] = new ShooterConfig(0,0, 0);
+    DistanceArray[1] = new ShooterConfig(0, 0, 0);
+    DistanceArray[2] = new ShooterConfig(0,0, 0);
+    DistanceArray[3] = new ShooterConfig(0,0, 0);
+    DistanceArray[4] = new ShooterConfig(0,0, 0);
+    DistanceArray[5] = new ShooterConfig(0,0, 0);
+    DistanceArray[6] = new ShooterConfig(0,0, 0);
+    DistanceArray[7] = new ShooterConfig(0,0, 0);
+    DistanceArray[8] = new ShooterConfig(0,0, 0);
+    DistanceArray[9] = new ShooterConfig(0,0, 0);
+    DistanceArray[10] = new ShooterConfig(0,0, 0);
+    DistanceArray[11] = new ShooterConfig(0,0, 0);
+    DistanceArray[12] = new ShooterConfig(0,0, 0);
+    DistanceArray[13] = new ShooterConfig(0,0, 0);
+    DistanceArray[14] = new ShooterConfig(0,0, 0);
+    DistanceArray[15] = new ShooterConfig(0,0, 0);
+    DistanceArray[16] = new ShooterConfig(0,0, 0);
+    DistanceArray[17] = new ShooterConfig(0,0, 0);
+    DistanceArray[18] = new ShooterConfig(0,0, 0);
+    DistanceArray[19] = new ShooterConfig(0,0, 0);
+    
+
 
-    KShooterController.setOutputRange(0, 1);
-    /*
-    hood_motorController = new MotorController("Hood", Constants.hoodID);
-    KHoodController = hood_motorController.getPID();
-    KHoodEncoder = shooter_motorController.getEncoder();*/
+
+
+    //KShooterController.setP(5e-4);
+    //KShooterController.setI(6e-7);
+    KShooterController.setIMaxAccum(0.9, 0);
+    //KShooterController.setD(0.0);
+
+    //hood_motorController = new MotorController("Hood", Constants.Shooter.hoodID,40,true);
+    //KHoodController = hood_motorController.getPID();
+    //KHoodEncoder = shooter_motorController.getEncoder();
 
   }
+  public void updatePID(){
 
+    
+    if (dashTunePid.getBoolean(false)){
+      
+      Pconstant = PID_P.getDouble(0);
+      Iconstant = PID_I.getDouble(0);
+      Dconstant = PID_D.getDouble(0);
+      Fconstant = PID_F.getDouble(0);
+      MaxOutputConstant = PID_MaxOutput.getDouble(0);
+      MinOutputConstant = PID_MinOutput.getDouble(0);
+      IMaxAccumconstant = PID_IMaxAccum.getDouble(0);
+      IMaxAccumIDconstant =(int)PID_IMaxAccumID.getDouble(0);
+      shooter_motorController.getPID().setOutputRange(MaxOutputConstant, MinOutputConstant);
+      shooter_motorController.getPID().setP(Pconstant);
+      shooter_motorController.getPID().setI(Iconstant);
+      shooter_motorController.getPID().setD(Dconstant);
+      shooter_motorController.getPID().setFF(Fconstant);
+      shooter_motorController.getPID().setIMaxAccum(IMaxAccumconstant, IMaxAccumIDconstant);
+    }
+  }
+  public double[] lookup(double Currentdistance){
+    if (Currentdistance < DistanceArray[0].getDistance()){
+      return DistanceArray[0].getVelocityAndAngle();
+    }
+    int i =0;
+    while(i<DistanceArray.length){
```

Comment:
Incrementing after check can cause a index out of range error. For loop might be better